strawberry/schema/schema_converter.py

```diff
@@ -222,11 +222,17 @@ def get_graphql_fields() -> Dict[str, GraphQLField]:
 
             return graphql_fields
 
+        def resolve_type(obj, info, type_):
+            # TODO: this will probably break when passing dicts
+            # or even non strawberry types
```

Comment:
Should this PR be resolved before merge? Or xfail tests added?

Reply:
We should add a test for sure :)